src/Microsoft.DotNet.XHarness.CLI/Program.cs

```diff
@@ -28,27 +28,18 @@ public static int Main(string[] args)
                 args[0] = "apple";
             }
 
-            // Root command: will use the platform specific commands to perform the appropriate action.
-            var commands = new CommandSet("xharness");
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
-            {
-                commands.Add(new AppleCommandSet());
-            }
-            else if (args[0] == "apple")
+            if (args[0] == "apple" && RuntimeInformation.IsOSPlatform(OSPlatform.OSX))

```

Comment:
This throws when it is just `dotnet xharness`:

```suggestion
            if (args.Length > 0 && args[0] == "apple" && !RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```